### PR TITLE
Move solved type classes into Prim modules & bump test deps

### PR DESCRIPTION
**Pull request by LiamGoodacre**
WIP
Can split this into two PRs if that makes more sense but things aren't going to work until the test deps are consistent.
Next I'll update our tests to switch from Eff to Effect.
Resolves https://github.com/purescript/purescript/issues/3179

### CODE REVIEW

**Comment by garyb** [comment]
I think doing it this way is fine 👍 I have a branch with updated dependencies too, but until the Prim/Prelude record instances are in that doesn't work anyway.

**Comment by kRITZCREEK** [comment]
I read the code that's there so far and didn't spot any obvious oversights 👍

File: src/Language/PureScript/Docs/Prim.hs [hunk 1]
```diff
@@ -273,22 +330,26 @@ partial = primClass "Partial" $ T.unlines
...
-fail = primClassOf (P.primSubName "TypeError") "Fail" $ T.unlines
-  [ "The Fail type class is part of the custom type errors feature. To provide"
-  , "a custom type error when someone tries to use a particular instance,"
-  , "write that instance out with a Fail constraint."
-  , ""
-  , "For more information, see"
-  , "[the Custom Type Errors guide](https://github.com/purescript/documentation/blob/master/guides/Custom-Type-Errors.md)."
+kindOrdering :: Declaration
+kindOrdering = primKindOf (P.primSubName "Ordering") "Ordering" $ T.unlines
+  [ "The `Ordering` kind represents the three possibilites of comparing twos"
```
Comment (LiamGoodacre) [inline src/Language/PureScript/Docs/Prim.hs]: Oops `twos` -> `two`

File: src/Language/PureScript/Environment.hs [hunk 2]
```diff
@@ -349,9 +365,9 @@ primKinds =
 primTypes :: M.Map (Qualified (ProperName 'TypeName)) (Kind, TypeKind)
 primTypes =
   M.fromList
```
Comment (LiamGoodacre) [inline src/Language/PureScript/Environment.hs]: Should have `Partial : kindConstraint` for consistency.

**Comment by LiamGoodacre** [comment]
Updated a whole bunch of tests. 😓

Getting cannot find `Prim.Symbol` and others like that, so I've probably missed updating something.
Will try investigating tomorrow evening.

Updated purescript-typelevel-prelude to reexport stuff from Prim submodules - but that needs more work and somewhat integrating with prelude.

**Comment by LiamGoodacre** [comment]
Also haven't fixed tests that involve ST/Ref as I'm not sure what we're doing there.

**Comment by garyb** [comment]
In what sense? The ST & Ref libraries have been updated (ST being its own monad now).

**Comment by LiamGoodacre** [comment]
Right! I'm being a moron :smile_cat:

**Comment by LiamGoodacre** [comment]
Think there was one that used `runPure`, so that will need rethinking.

**Comment by LiamGoodacre** [comment]
So there's this test: `examples/failing/Eff.purs`

```
-- @shouldFailWith TypesDoNotUnify
module Main where

import Prelude
import Control.Monad.Eff
import Control.Monad.ST
import Control.Monad.Eff.Console

test = pureST (do
         ref <- newSTRef 0
         log "ST"
         modifySTRef ref $ \n -> n + 1
         readSTRef ref)
```

Which I believe was testing that you can't log in a pure ST.

Thoughts on what this should change to?

**Comment by LiamGoodacre** [comment]
Similarly the passing one: `examples/passing/Eff.purs`
```purescript
module Main where

import Prelude
import Control.Monad.Eff
import Control.Monad.ST
import Control.Monad.Eff.Console (log, logShow)

test1 = do
  log "Line 1"
  log "Line 2"

test2 = runPure (runST (do
          ref <- newSTRef 0.0
          _ <- modifySTRef ref $ \n -> n + 1.0
          readSTRef ref))

test3 = pureST (do
          ref <- newSTRef 0.0
          _ <- modifySTRef ref $ \n -> n + 1.0
          readSTRef ref)

main = do
  test1
  logShow test2
  logShow test3
  log "Done"
```

Wondering if these tests should just be deleted, given their focus is actually on `Eff`.

**Comment by garyb** [comment]
> Wondering if these tests should just be deleted, given their focus is actually on `Eff`.

Yeah that sounds reasonable to me. We probably have quite a lot of redundant tests, but no harm having them unless like this they don't make sense anymore.

**Comment by LiamGoodacre** [comment]
Thanks @kRITZCREEK for finishing off fixing up the tests!

**Pull request merged**